kv/kvserver: TestRemovePlaceholderRace failed

[(kv/kvserver).TestRemovePlaceholderRace failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1988944&tab=buildLog) on [master@82f172ab2c717ffc87c15686e2f438392a3b522f](https://github.com/cockroachdb/cockroach/commits/82f172ab2c717ffc87c15686e2f438392a3b522f):

Fatal error:
```
panic: timed out during shutdown [recovered]
	panic: timed out during shutdown [recovered]
	panic: timed out during shutdown

```

Stack:
```
goroutine 172093 [running]:
testing.tRunner.func1(0xc00ebb6f00)
	/usr/local/go/src/testing/testing.go:874 +0x69f
panic(0x6349d00, 0x7a696a0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:99 +0x27f
panic(0x6349d00, 0x7a696a0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.(*multiTestContext).Stop(0xc007110c40)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_test.go:467 +0x3d0
github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestRemovePlaceholderRace(0xc00ebb6f00)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_raft_test.go:2759 +0x658
testing.tRunner(0xc00ebb6f00, 0x6b34990)
	/usr/local/go/src/testing/testing.go:909 +0x19a
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:960 +0x652

```

<details><summary>Log preceding fatal error</summary><p>

```
I200604 19:35:42.890552 219369 gossip/client.go:129  [n2] closing client to n1 (127.0.0.1:44493): stopping outgoing client to n1 (127.0.0.1:44493); already have incoming
W200604 19:35:43.025817 172643 kv/kvserver/store.go:1652  [s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200604 19:35:43.637341 219765 gossip/client.go:124  [n3] started gossip client to 127.0.0.1:44493
I200604 19:35:43.652840 219765 gossip/client.go:129  [n3] closing client to n1 (127.0.0.1:44493): stopping outgoing client to n1 (127.0.0.1:44493); already have incoming
W200604 19:35:43.716997 172453 kv/kvserver/store.go:1652  [s2,r1/103:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200604 19:35:43.863691 219838 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:44493
W200604 19:35:43.866052 172245 kv/kvserver/store.go:1652  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200604 19:35:43.874961 219838 gossip/client.go:129  [n2] closing client to n1 (127.0.0.1:44493): stopping outgoing client to n1 (127.0.0.1:44493); already have incoming
W200604 19:35:44.056792 172643 kv/kvserver/store.go:1652  [s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200604 19:35:44.629414 220250 gossip/client.go:124  [n3] started gossip client to 127.0.0.1:33593
I200604 19:35:44.642271 220250 gossip/client.go:129  [n3] closing client to n2 (127.0.0.1:33593): stopping outgoing client to n2 (127.0.0.1:33593); already have incoming
W200604 19:35:44.690371 172453 kv/kvserver/store.go:1652  [s2,r1/103:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200604 19:35:44.868519 220378 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:44493
I200604 19:35:44.874244 220378 gossip/client.go:129  [n2] closing client to n1 (127.0.0.1:44493): stopping outgoing client to n1 (127.0.0.1:44493); already have incoming
W200604 19:35:44.951044 172245 kv/kvserver/store.go:1652  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200604 19:35:45.043634 172643 kv/kvserver/store.go:1652  [s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200604 19:35:45.629325 220844 gossip/client.go:124  [n3] started gossip client to 127.0.0.1:44493
I200604 19:35:45.636665 220844 gossip/client.go:129  [n3] closing client to n1 (127.0.0.1:44493): stopping outgoing client to n1 (127.0.0.1:44493); already have incoming
W200604 19:35:45.696112 172453 kv/kvserver/store.go:1652  [s2,r1/103:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200604 19:35:45.865861 221047 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:44493
I200604 19:35:45.881975 221047 gossip/client.go:129  [n2] closing client to n1 (127.0.0.1:44493): stopping outgoing client to n1 (127.0.0.1:44493); already have incoming
W200604 19:35:46.036361 172643 kv/kvserver/store.go:1652  [s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200604 19:35:46.089048 172245 kv/kvserver/store.go:1652  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200604 19:35:46.592105 172453 kv/kvserver/store.go:1652  [s2,r1/103:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200604 19:35:46.638889 221416 gossip/client.go:124  [n3] started gossip client to 127.0.0.1:44493
I200604 19:35:46.645442 221416 gossip/client.go:129  [n3] closing client to n1 (127.0.0.1:44493): stopping outgoing client to n1 (127.0.0.1:44493); already have incoming
I200604 19:35:46.867429 221520 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:44493
I200604 19:35:46.876507 221520 gossip/client.go:129  [n2] closing client to n1 (127.0.0.1:44493): stopping outgoing client to n1 (127.0.0.1:44493); already have incoming
W200604 19:35:47.011311 172245 kv/kvserver/store.go:1652  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200604 19:35:47.080788 172643 kv/kvserver/store.go:1652  [s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200604 19:35:47.620699 172453 kv/kvserver/store.go:1652  [s2,r1/103:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200604 19:35:47.649607 221930 gossip/client.go:124  [n3] started gossip client to 127.0.0.1:33593
I200604 19:35:47.657021 221930 gossip/client.go:129  [n3] closing client to n2 (127.0.0.1:33593): stopping outgoing client to n2 (127.0.0.1:33593); already have incoming
I200604 19:35:47.871477 222053 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:44493
I200604 19:35:47.886976 222053 gossip/client.go:129  [n2] closing client to n1 (127.0.0.1:44493): stopping outgoing client to n1 (127.0.0.1:44493); already have incoming
W200604 19:35:47.985690 172643 kv/kvserver/store.go:1652  [s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200604 19:35:48.091361 172245 kv/kvserver/store.go:1652  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200604 19:35:48.659820 222514 gossip/client.go:124  [n3] started gossip client to 127.0.0.1:44493
W200604 19:35:48.665160 172453 kv/kvserver/store.go:1652  [s2,r1/103:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200604 19:35:48.680055 222514 gossip/client.go:129  [n3] closing client to n1 (127.0.0.1:44493): stopping outgoing client to n1 (127.0.0.1:44493); already have incoming
I200604 19:35:48.869992 222512 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:44493
I200604 19:35:48.880360 222512 gossip/client.go:129  [n2] closing client to n1 (127.0.0.1:44493): stopping outgoing client to n1 (127.0.0.1:44493); already have incoming
W200604 19:35:49.000659 172245 kv/kvserver/store.go:1652  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200604 19:35:49.079473 172643 kv/kvserver/store.go:1652  [s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200604 19:35:49.568270 172453 kv/kvserver/store.go:1652  [s2,r1/103:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200604 19:35:49.660847 222897 gossip/client.go:124  [n3] started gossip client to 127.0.0.1:44493
I200604 19:35:49.688002 222897 gossip/client.go:129  [n3] closing client to n1 (127.0.0.1:44493): stopping outgoing client to n1 (127.0.0.1:44493); already have incoming
I200604 19:35:49.876706 223111 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:44493
I200604 19:35:49.884259 223111 gossip/client.go:129  [n2] closing client to n1 (127.0.0.1:44493): stopping outgoing client to n1 (127.0.0.1:44493); already have incoming
--- FAIL: TestRemovePlaceholderRace (105.94s)

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRemovePlaceholderRace PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRemovePlaceholderRace.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestRemovePlaceholderRace failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1990252&tab=buildLog) on [master@c51b24908aca978835cacba74ad0866c2506fe03](https://github.com/cockroachdb/cockroach/commits/c51b24908aca978835cacba74ad0866c2506fe03):

Fatal error:
```
panic: timed out during shutdown [recovered]
	panic: timed out during shutdown [recovered]
	panic: timed out during shutdown

```

Stack:
```
goroutine 173489 [running]:
testing.tRunner.func1(0xc01069d600)
	/usr/local/go/src/testing/testing.go:874 +0x69f
panic(0x5a916a0, 0x71bfac0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:99 +0x27f
panic(0x5a916a0, 0x71bfac0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.(*multiTestContext).Stop(0xc005d38380)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_test.go:467 +0x3d0
github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestRemovePlaceholderRace(0xc01069d600)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_raft_test.go:2759 +0x658
testing.tRunner(0xc01069d600, 0x627f420)
	/usr/local/go/src/testing/testing.go:909 +0x19a
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:960 +0x652

```

<details><summary>Log preceding fatal error</summary><p>

```
W200605 06:21:56.686162 173706 kv/kvserver/store.go:1652  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200605 06:21:56.817307 222275 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:46675
I200605 06:21:56.840495 222275 gossip/client.go:129  [n2] closing client to n1 (127.0.0.1:46675): stopping outgoing client to n1 (127.0.0.1:46675); already have incoming
W200605 06:21:56.859391 173910 kv/kvserver/store.go:1652  [s2,r1/103:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200605 06:21:57.135468 222467 gossip/client.go:124  [n3] started gossip client to 127.0.0.1:45917
I200605 06:21:57.159359 222467 gossip/client.go:129  [n3] closing client to n2 (127.0.0.1:45917): stopping outgoing client to n2 (127.0.0.1:45917); already have incoming
W200605 06:21:57.497952 174089 kv/kvserver/store.go:1652  [s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200605 06:21:57.809533 173706 kv/kvserver/store.go:1652  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200605 06:21:57.830590 222801 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:46675
I200605 06:21:57.836113 222801 gossip/client.go:129  [n2] closing client to n1 (127.0.0.1:46675): stopping outgoing client to n1 (127.0.0.1:46675); already have incoming
W200605 06:21:57.945067 173910 kv/kvserver/store.go:1652  [s2,r1/103:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200605 06:21:57.953269 173630 kv/kvserver/queue.go:580  [s1] rate limited in MaybeAdd (gc): node unavailable; try another peer
I200605 06:21:57.953923 173630 kv/kvserver/queue.go:580  [s1] rate limited in MaybeAdd (merge): node unavailable; try another peer
I200605 06:21:57.954453 173630 kv/kvserver/queue.go:580  [s1] rate limited in MaybeAdd (split): node unavailable; try another peer
I200605 06:21:57.954904 173630 kv/kvserver/queue.go:580  [s1] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I200605 06:21:57.955296 173630 kv/kvserver/queue.go:580  [s1] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I200605 06:21:57.955587 173630 kv/kvserver/queue.go:580  [s1] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
I200605 06:21:57.955874 173630 kv/kvserver/queue.go:580  [s1] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I200605 06:21:57.956155 173630 kv/kvserver/queue.go:580  [s1] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I200605 06:21:58.140833 223014 gossip/client.go:124  [n3] started gossip client to 127.0.0.1:46675
I200605 06:21:58.150429 223014 gossip/client.go:129  [n3] closing client to n1 (127.0.0.1:46675): stopping outgoing client to n1 (127.0.0.1:46675); already have incoming
W200605 06:21:58.506527 174089 kv/kvserver/store.go:1652  [s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200605 06:21:58.707873 173706 kv/kvserver/store.go:1652  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200605 06:21:58.830947 223373 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:46675
I200605 06:21:58.836442 223373 gossip/client.go:129  [n2] closing client to n1 (127.0.0.1:46675): stopping outgoing client to n1 (127.0.0.1:46675); already have incoming
W200605 06:21:58.997887 173910 kv/kvserver/store.go:1652  [s2,r1/103:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200605 06:21:59.152539 223563 gossip/client.go:124  [n3] started gossip client to 127.0.0.1:46675
I200605 06:21:59.159265 223563 gossip/client.go:129  [n3] closing client to n1 (127.0.0.1:46675): stopping outgoing client to n1 (127.0.0.1:46675); already have incoming
W200605 06:21:59.531380 174089 kv/kvserver/store.go:1652  [s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200605 06:21:59.723363 173706 kv/kvserver/store.go:1652  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200605 06:21:59.832212 223884 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:46675
I200605 06:21:59.838573 223884 gossip/client.go:129  [n2] closing client to n1 (127.0.0.1:46675): stopping outgoing client to n1 (127.0.0.1:46675); already have incoming
W200605 06:21:59.956370 173910 kv/kvserver/store.go:1652  [s2,r1/103:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200605 06:22:00.163321 224142 gossip/client.go:124  [n3] started gossip client to 127.0.0.1:45917
I200605 06:22:00.167526 224142 gossip/client.go:129  [n3] closing client to n2 (127.0.0.1:45917): stopping outgoing client to n2 (127.0.0.1:45917); already have incoming
W200605 06:22:00.672441 174089 kv/kvserver/store.go:1652  [s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200605 06:22:00.703561 173706 kv/kvserver/store.go:1652  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200605 06:22:00.833306 224674 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:46675
I200605 06:22:00.837457 224674 gossip/client.go:129  [n2] closing client to n1 (127.0.0.1:46675): stopping outgoing client to n1 (127.0.0.1:46675); already have incoming
W200605 06:22:00.885650 173910 kv/kvserver/store.go:1652  [s2,r1/103:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200605 06:22:01.152569 224931 gossip/client.go:124  [n3] started gossip client to 127.0.0.1:46675
I200605 06:22:01.158119 224931 gossip/client.go:129  [n3] closing client to n1 (127.0.0.1:46675): stopping outgoing client to n1 (127.0.0.1:46675); already have incoming
W200605 06:22:01.666629 174089 kv/kvserver/store.go:1652  [s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200605 06:22:01.771401 173706 kv/kvserver/store.go:1652  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200605 06:22:01.834744 225490 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:46675
I200605 06:22:01.839515 225490 gossip/client.go:129  [n2] closing client to n1 (127.0.0.1:46675): stopping outgoing client to n1 (127.0.0.1:46675); already have incoming
W200605 06:22:01.866622 173910 kv/kvserver/store.go:1652  [s2,r1/103:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200605 06:22:02.153170 225638 gossip/client.go:124  [n3] started gossip client to 127.0.0.1:46675
I200605 06:22:02.158389 225638 gossip/client.go:129  [n3] closing client to n1 (127.0.0.1:46675): stopping outgoing client to n1 (127.0.0.1:46675); already have incoming
--- FAIL: TestRemovePlaceholderRace (95.37s)

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=2

```
make stressrace TESTS=TestRemovePlaceholderRace PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRemovePlaceholderRace.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestRemovePlaceholderRace failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2009830&tab=buildLog) on [master@72d1ec4b9974c8ecd755f43c77448b59ae6706d9](https://github.com/cockroachdb/cockroach/commits/72d1ec4b9974c8ecd755f43c77448b59ae6706d9):

Fatal error:
```
panic: timed out during shutdown [recovered]
	panic: timed out during shutdown [recovered]
	panic: timed out during shutdown

```

Stack:
```
goroutine 175796 [running]:
testing.tRunner.func1(0xc007731600)
	/usr/local/go/src/testing/testing.go:874 +0x69f
panic(0x6517640, 0x7d685c0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:99 +0x27f
panic(0x6517640, 0x7d685c0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.(*multiTestContext).Stop(0xc000ad8fc0)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_test.go:479 +0x3d0
github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestRemovePlaceholderRace(0xc007731600)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_raft_test.go:2759 +0x658
testing.tRunner(0xc007731600, 0x6d378c8)
	/usr/local/go/src/testing/testing.go:909 +0x19a
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:960 +0x652

```

<details><summary>Log preceding fatal error</summary><p>

```
I200613 01:56:50.226822 223595 gossip/client.go:124  [n3] started gossip client to 127.0.0.1:40189
I200613 01:56:50.249625 223595 gossip/client.go:129  [n3] closing client to n2 (127.0.0.1:40189): stopping outgoing client to n2 (127.0.0.1:40189); already have incoming
W200613 01:56:50.419546 175886 kv/kvserver/store.go:1651  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200613 01:56:50.456326 175887 kv/kvserver/store.go:1651  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: node unavailable; try another peer
W200613 01:56:50.520502 176299 kv/kvserver/store.go:1651  [s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200613 01:56:50.780738 176074 kv/kvserver/store.go:1651  [s2,r1/103:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200613 01:56:50.801418 223860 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:37497
I200613 01:56:50.809169 223860 gossip/client.go:129  [n2] closing client to n1 (127.0.0.1:37497): stopping outgoing client to n1 (127.0.0.1:37497); already have incoming
W200613 01:56:51.016133 175888 kv/kvserver/store.go:1651  [s1,r2/1:/System/NodeLiveness{-Max}] could not gossip node liveness: node unavailable; try another peer
I200613 01:56:51.228836 224018 gossip/client.go:124  [n3] started gossip client to 127.0.0.1:37497
I200613 01:56:51.246069 224018 gossip/client.go:129  [n3] closing client to n1 (127.0.0.1:37497): stopping outgoing client to n1 (127.0.0.1:37497); already have incoming
W200613 01:56:51.351945 175887 kv/kvserver/store.go:1651  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: node unavailable; try another peer
W200613 01:56:51.410963 175886 kv/kvserver/store.go:1651  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200613 01:56:51.420644 176299 kv/kvserver/store.go:1651  [s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200613 01:56:51.804170 224244 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:37497
I200613 01:56:51.836853 224244 gossip/client.go:129  [n2] closing client to n1 (127.0.0.1:37497): stopping outgoing client to n1 (127.0.0.1:37497); already have incoming
W200613 01:56:51.882096 176074 kv/kvserver/store.go:1651  [s2,r1/103:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200613 01:56:51.967490 175888 kv/kvserver/store.go:1651  [s1,r2/1:/System/NodeLiveness{-Max}] could not gossip node liveness: node unavailable; try another peer
I200613 01:56:52.008682 175906 kv/kvserver/queue.go:580  [s1] rate limited in MaybeAdd (gc): node unavailable; try another peer
I200613 01:56:52.009404 175906 kv/kvserver/queue.go:580  [s1] rate limited in MaybeAdd (merge): node unavailable; try another peer
I200613 01:56:52.009813 175906 kv/kvserver/queue.go:580  [s1] rate limited in MaybeAdd (split): node unavailable; try another peer
I200613 01:56:52.010407 175906 kv/kvserver/queue.go:580  [s1] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I200613 01:56:52.010726 175906 kv/kvserver/queue.go:580  [s1] rate limited in MaybeAdd (replicaGC): node unavailable; try another peer
I200613 01:56:52.011155 175906 kv/kvserver/queue.go:580  [s1] rate limited in MaybeAdd (raftlog): node unavailable; try another peer
I200613 01:56:52.011842 175906 kv/kvserver/queue.go:580  [s1] rate limited in MaybeAdd (raftsnapshot): node unavailable; try another peer
I200613 01:56:52.012400 175906 kv/kvserver/queue.go:580  [s1] rate limited in MaybeAdd (consistencyChecker): node unavailable; try another peer
I200613 01:56:52.248627 224301 gossip/client.go:124  [n3] started gossip client to 127.0.0.1:37497
W200613 01:56:52.251871 175887 kv/kvserver/store.go:1651  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: node unavailable; try another peer
I200613 01:56:52.258342 224301 gossip/client.go:129  [n3] closing client to n1 (127.0.0.1:37497): stopping outgoing client to n1 (127.0.0.1:37497); already have incoming
W200613 01:56:52.297122 175886 kv/kvserver/store.go:1651  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200613 01:56:52.467551 176299 kv/kvserver/store.go:1651  [s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200613 01:56:52.806427 224575 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:37497
I200613 01:56:52.818087 224575 gossip/client.go:129  [n2] closing client to n1 (127.0.0.1:37497): stopping outgoing client to n1 (127.0.0.1:37497); already have incoming
W200613 01:56:52.963459 175888 kv/kvserver/store.go:1651  [s1,r2/1:/System/NodeLiveness{-Max}] could not gossip node liveness: node unavailable; try another peer
W200613 01:56:52.992390 176074 kv/kvserver/store.go:1651  [s2,r1/103:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200613 01:56:53.152202 175887 kv/kvserver/store.go:1651  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: node unavailable; try another peer
I200613 01:56:53.255841 224835 gossip/client.go:124  [n3] started gossip client to 127.0.0.1:40189
I200613 01:56:53.262195 224835 gossip/client.go:129  [n3] closing client to n2 (127.0.0.1:40189): stopping outgoing client to n2 (127.0.0.1:40189); already have incoming
W200613 01:56:53.321732 175886 kv/kvserver/store.go:1651  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200613 01:56:53.626407 176299 kv/kvserver/store.go:1651  [s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200613 01:56:53.809059 224972 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:37497
I200613 01:56:53.816718 224972 gossip/client.go:129  [n2] closing client to n1 (127.0.0.1:37497): stopping outgoing client to n1 (127.0.0.1:37497); already have incoming
W200613 01:56:53.825354 175888 kv/kvserver/store.go:1651  [s1,r2/1:/System/NodeLiveness{-Max}] could not gossip node liveness: node unavailable; try another peer
W200613 01:56:53.941088 176074 kv/kvserver/store.go:1651  [s2,r1/103:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200613 01:56:54.107251 175887 kv/kvserver/store.go:1651  [s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: node unavailable; try another peer
I200613 01:56:54.258298 225258 gossip/client.go:124  [n3] started gossip client to 127.0.0.1:37497
I200613 01:56:54.265998 225258 gossip/client.go:129  [n3] closing client to n1 (127.0.0.1:37497): stopping outgoing client to n1 (127.0.0.1:37497); already have incoming
W200613 01:56:54.472787 175886 kv/kvserver/store.go:1651  [s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200613 01:56:54.563395 176299 kv/kvserver/store.go:1651  [s3,r1/3:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
--- FAIL: TestRemovePlaceholderRace (130.08s)

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestRemovePlaceholderRace PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRemovePlaceholderRace.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/kv/kvserver/replica_consistency.go
+++ b/pkg/kv/kvserver/replica_consistency.go
@@ -425,9 +425,9 @@ func (r *Replica) getChecksum(ctx context.Context, id uuid.UUID) (ReplicaChecksu
 	r.mu.Unlock()
 	// Wait
 	select {
-	case <-r.store.Stopper().ShouldStop():
+	case <-r.store.Stopper().ShouldQuiesce():
 		return ReplicaChecksum{},
-			errors.Errorf("store has stopped while waiting for compute checksum (ID = %s)", id)
+			errors.Errorf("store quiescing while waiting for compute checksum (ID = %s)", id)
 	case <-ctx.Done():
 		return ReplicaChecksum{},
 			errors.Wrapf(ctx.Err(), "while waiting for compute checksum (ID = %s)", id)

